src/override/malloc.cc

```diff
@@ -104,12 +104,33 @@ extern "C"
   }
 #endif
 
-  SNMALLOC_EXPORT void*
-    SNMALLOC_NAME_MANGLE(aligned_alloc)(size_t alignment, size_t size)
+  inline size_t aligned_size(size_t alignment, size_t size)
   {
-    assert((size % alignment) == 0);
-    (void)alignment;
-    return SNMALLOC_NAME_MANGLE(malloc)(size);
+    // Client responsible for checking alignment is not zero
+    // Client responsible for checking for overflow
+    assert((size + alignment) > size);
```

Comment:
Why are we checking for overflow on this addition?

Reply:
Not sure.  It was in the previous implementation, but seems unnecessary now.  I think it may be from your original version, and we should have dropped it long ago.